### PR TITLE
fix for #3136 Temporary workflow xml files are not deleted

### PR DESCRIPTION
**Pull request by fviale**
- rationalize all usages of FileInputStream/OutputStream/Reader/Writer with try-with-resources
 - ServerJobAndTaskLogs: delete temporary html file produced by WorkflowSubmitter when a job is deleted

### CODE REVIEW

**Comment by fviale** [comment]
Retest this please

File: rest/rest-api/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/utils/Zipper.java [hunk 1]
```diff
@@ -160,7 +170,9 @@ public static void unzip(InputStream is, File outFile) throws IOException {
                         entryContainer.mkdirs();
                     }
                     if (!entryFile.isDirectory()) {
-                        Zipper.ZIP.unzipEntry(zis, new FileOutputStream(entryFile));
+                        FileOutputStream outputStream = new FileOutputStream(entryFile);
```
Comment (marcocast) [inline rest/rest-api/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/utils/Zipper.java]: don't we need to close this outputStream ?

File: rest/rest-api/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/utils/Zipper.java [hunk 2]
```diff
@@ -119,10 +125,14 @@ private static void zipFiles(List<File> files, String basepath, OutputStream os)
                 closer.register(zos);
                 if (files.size() == 1) {
                     File file = files.get(0);
-                    writeZipEntry(new ZipEntry(file.getName()), new FileInputStream(file), zos);
+                    FileInputStream inputStream = new FileInputStream(file);
```
Comment (marcocast) [inline rest/rest-api/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/utils/Zipper.java]: don't we need to close this inputStream ?

Comment (fviale) [reply rest/rest-api/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/utils/Zipper.java]: this is done by the "closer"

File: rest/rest-api/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/utils/Zipper.java (hunk 1 above)
Comment (fviale) [reply rest/rest-api/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/utils/Zipper.java]: Same remark

**Pull request merged**